src/functional/ffront/decorator.py

```diff
@@ -94,6 +94,26 @@ def _filter_closure_vars_by_type(closure_vars: dict[str, Any], *types: type) ->
     return {name: value for name, value in closure_vars.items() if isinstance(value, types)}
 
 
+def _canonicalize_args(node_params, args, kwargs) -> tuple[tuple, dict]:
+    new_args = []
+
+    for param_i, param in enumerate(node_params):
+        if param.id in kwargs:
+            if param.id not in ["out", "domain"] and param_i < len(args):
+                raise ProgramTypeError(f"got multiple values for argument {param.id}.")
+            new_args.append(kwargs[param.id])
+            kwargs.pop(param.id)
+        elif param_i < len(args):
+            new_args.append(args[param_i])
+        else:
+            # case when parameter is in function definition but not in function call
+            # e.g. function definition includes 3 parameters, but only 2 were given as inputs
```

Comment:
```suggestion
            # e.g. function expects 3 parameters, but only 2 were given. To avoid duplication of
            # error handling this is covered later in `accept_args`.
```
Nitpicking.